### PR TITLE
Add multinode.yml workflow

### PR DESCRIPTION
**Pull request by markgoddard**
- **Add CODEOWNERS file**
- **Add a reusable GitHub workflow to deploy a multinode test cluster**
- **Add multinode.yml workflow**

### CODE REVIEW

File: .github/workflows/multinode.yml [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+      os_distribution:
+        description: Host OS distribution
+        type: string
+        default: rocky
+      os_release:
+        description: Host OS release
+        type: string
+        default: '9'
+      ssh_username:
+        description: User for terraform to access the all-in-one VM
```
Comment (Alex-Welsh) [inline .github/workflows/multinode.yml]:
```suggestion
        description: User for terraform to access the Multinode VMs
```

File: .github/workflows/multinode.yml [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        type: string
+        default: cloud-user
+      neutron_plugin:
+        description: Neutron ML2 plugin
+        type: string
+        default: ovn
+      stackhpc_kayobe_config_version:
```
Comment (Alex-Welsh) [inline .github/workflows/multinode.yml:42]: required?

File: .github/workflows/multinode.yml [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+        working-directory: ${{ github.workspace }}/terraform-kayobe-multinode
+
+      - name: Configure Ansible
+        run: |
+          echo '${{ env.KAYOBE_VAULT_PASSWORD }}' > vault-pw
+
+          cat << EOF >> ansible/vars/defaults.yml
+          kayobe_config_version: ${{ inputs.upgrade && inputs.stackhpc_kayobe_config_previous_version || inputs.stackhpc_kayobe_config_version }}
+          ssh_key_path: ${{ github.workspace }}/terraform-kayobe-multinode/id_rsa
+          vxlan_vni: 53299
```
Comment (Alex-Welsh) [inline .github/workflows/multinode.yml]: Would be nice if this was randomised/customisable, perhaps in some bounded range for "CI multinodes"

File: .github/workflows/multinode.yml [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        id: image_name
+        run: |
+          if [ -z "${{ inputs.multinode_image_override }}" ]; then
+            echo image_name=overcloud-${{ inputs.os_distribution }}-${{ inputs.os_release }}-${{ steps.image_tag.outputs.image_tag }} >> $GITHUB_OUTPUT
+          else
+            echo image_name=${{ inputs.multinode_image_override }} >> $GITHUB_OUTPUT
+          fi
+
+      - name: Install terraform
+        uses: hashicorp/setup-terraform@v2
```
Comment (Alex-Welsh) [inline .github/workflows/multinode.yml:135]:
```suggestion
        uses: hashicorp/setup-terraform@v3
```

File: .github/workflows/multinode.yml [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+        type: string
+      terraform_kayobe_multinode_version:
+        description: terraform-kayobe-multinode version
+        type: string
+        default: main
+      upgrade:
+        description: Whether to perform an upgrade
+        type: boolean
+        default: false
+      break_on:
```
Comment (Alex-Welsh) [inline .github/workflows/multinode.yml:57]: might be better as a choice rather than a string

File: .github/workflows/multinode.yml [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+
+      - name: Upload test result artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-results-multinode-${{ inputs.os_distribution }}-${{ inputs.os_release }}-${{ inputs.neutron_plugin }}${{ inputs.upgrade && '-upgrade' || '' }}
+          path: |
+            ${{ github.workspace }}/logs/
+        if: ${{ always() && steps.config_ach.outcome == 'success' }}
+
+      - name: Break on failure
```
Comment (Alex-Welsh) [inline .github/workflows/multinode.yml:363]: Also worth having break on success/always?

**Review by Alex-Welsh — CHANGES_REQUESTED** [review]

File: .github/workflows/multinode.yml (hunk 5 above)
Comment (Alex-Welsh) [reply .github/workflows/multinode.yml:57]: I've now seen that the workflow calling this one has a choice, which makes sense

File: .github/workflows/multinode.yml (hunk 2 above)
Comment (markgoddard) [reply .github/workflows/multinode.yml:42]: +1

File: .github/workflows/multinode.yml (hunk 4 above)
Comment (markgoddard) [reply .github/workflows/multinode.yml:135]: It doesn't work for some reason. Perhaps the github runner container image?

File: .github/workflows/multinode.yml (hunk 3 above)
Comment (markgoddard) [reply .github/workflows/multinode.yml]: Oops, yes this should not be constant.

File: .github/workflows/multinode.yml (hunk 6 above)
Comment (markgoddard) [reply .github/workflows/multinode.yml:363]: +1

File: .github/workflows/multinode.yml [hunk 7]
```diff
@@ -0,0 +1,381 @@
...
+      - name: Terraform Plan
+        run: terraform plan -input=false
+        working-directory: ${{ github.workspace }}/terraform-kayobe-multinode
+        env:
+          OS_CLOUD: ${{ vars.OS_CLOUD }}
+          OS_APPLICATION_CREDENTIAL_ID: ${{ secrets.OS_APPLICATION_CREDENTIAL_ID }}
+          OS_APPLICATION_CREDENTIAL_SECRET: ${{ secrets.OS_APPLICATION_CREDENTIAL_SECRET }}
+
+      - name: Terraform Apply
+        id: tf_apply
```
Comment (Alex-Welsh) [inline .github/workflows/multinode.yml]: nit: unused

**Review by Alex-Welsh — DISMISSED** [review]

**Pull request merged**